src/adapters/localStorage.js

```diff
@@ -0,0 +1,136 @@
+"use strict";
+
+import BaseAdapter from "./base.js";
+
+const root = typeof window === "object" ? window : global;
+
+// Only load localStorage in a nodejs environment
+if (!root.hasOwnProperty("localStorage")) {
+  root.localStorage = require("localStorage");
+}
+
+export default class LocalStorage extends BaseAdapter {
+  constructor(dbname) {
+    super();
+    this._db = null;
+    // public properties
+    this.dbname = dbname;
+    this._keyStoreName = `${this.dbname}/__keys`;
+  }
+
+  _handleError(method) {
+    return err => {
+      const error = new Error(method + "() " + err.message);
+      error.stack = err.stack;
+      throw error;
+    };
+  }
+
+  get keys() {
+    return JSON.parse(localStorage.getItem(this._keyStoreName)) || [];
+  }
+
+  set keys(keys) {
+    localStorage.setItem(this._keyStoreName, JSON.stringify(keys));
+  }
+
+  /**
+   * Deletes every records in the current collection.
+   *
+   * @return {Promise}
+   */
+  clear() {
+    try {
+      localStorage.clear();
+      return Promise.resolve();
+    } catch (err) {
+      return Promise.reject(err)
+    }
+  }
+
+  /**
+   * Adds a record to the localStorage datastore.
+   *
+   * Note: An id value is required.
+   *
+   * @param  {Object} record The record object, including an id.
+   * @return {Promise}
+   */
+  create(record) {
+    if (this.keys.indexOf(record.id) !== -1)
+      return Promise.reject(new Error("Exists."))
+    localStorage.setItem(`${this.dbname}/${record.id}`, JSON.stringify(record));
+    this.keys = this.keys.concat(record.id);
+    return Promise.resolve(record);
+  }
+
+  /**
+   * Updates a record from the localStorage datastore.
+   *
+   * @param  {Object} record
+   * @return {Promise}
+   */
+  update(record) {
+    if (this.keys.indexOf(record.id) === -1)
+      return Promise.reject(new Error("Doesn't exist."))
+    localStorage.setItem(`${this.dbname}/${record.id}`, JSON.stringify(record));
+    return Promise.resolve(record);
+  }
+
+  /**
+   * Retrieve a record by its primary key from the localStorage datastore.
+   *
+   * @param  {String} id The record id.
+   * @return {Promise}
+   */
+  get(id) {
+    return Promise.resolve(
+      JSON.parse(localStorage.getItem(`${this.dbname}/${id}`)) || undefined);
```

Comment:
Can ids be any type of value? (asking for the formatting, in order to avoid collision of names